### DISCUSSION TITLE
Webhooks & Subscriptions

### DISCUSSION THREAD

**Discussion by rrrnld** [saleor/saleor#14194] [Q&A] [answered]
What exactly is the relationship between webhooks and subscriptions? 

My understanding is: "Normal" webhooks allow you to subscribe to certain events and receive a default payload as per the subscribed event type. Subscriptions allow you to do the same + alter the payload. Is this correct so far?

A team mate seems to have read somewhere that subscriptions will supersede normal webhooks, but I couldn't find anything about this. Is this planned?

And lastly: I think I have a buggy webhook in my self-hosted saleor database that is the result of some early experimentation. The `webhook_webhook` table contains the webhook with some GraphQL query in `subscription_query`:

``` graphql
subscription {
  event {
    ... on OrderConfirmed {
      __typename
      order {
        id
      }
    }
    ... on OrderCancelled {
      __typename
      order {
        id
      }
    }
    ... on OrderFulfilled {
      __typename
      order {
        id
      }
    }
    ... on FulfillmentCreated {
      __typename
      order {
        id
      }
      fulfillment {
        id
      }
    }
    ... on FulfillmentCanceled {
      __typename
      order {
        id
      }
      fulfillment {
        id
      }
    }
    ... on InvoiceRequested {
      __typename
      invoiceOrder: order {
        id
      }
    }
  }
}
````

There are also some entries in `webhook_webhookevents`:

```
+----+--------------------------+------------+
| id | event_type               | webhook_id |
|----+--------------------------+------------|
| 61 | order_fulfilled          | 1          |
| 62 | fulfillment_created      | 1          |
| 63 | notify_user              | 1          |
| 64 | fulfillment_canceled     | 1          |
| 65 | order_fully_paid         | 1          |
| 66 | invoice_requested        | 1          |
| 67 | order_cancelled          | 1          |
| 68 | order_confirmed          | 1          |
| 69 | order_created            | 1          |
| 70 | order_updated            | 1          |
+----+--------------------------+------------+
```

This seems in part redundant and in part contradictory -- is this correct? Do these WebhookEvents get created also for subscription-based webhooks?

Thanks in advance for any help & guidance!

**Comment by timuric** [comment, marked as answer]
Your reasoning is correct we recommend to always use subscriptions which improves performance and saves resources. We will deprecate standard webhooks at some point.

If you believe there is a bug please file an issue with reproduction steps and unexpected outcome.
